### ISSUE TITLE
Consider allowing developers to select a the length of a field in the Additional Fields API

### ISSUE THREAD

**Issue by senadir** [woothemes/woocommerce#42088]
Certain fields in Checkout would render full width, while others would render half width. In Checkout, the last field that's alone in a form will always render full width, while having 2 fields would make them render as halves.

We should look into a way to allow developers to indicate that a field should be full width, with the default is taking whatever is available. Ideally this option should be semantic in its meaning as it's a UI value in what should be a UI-independent definition.

**Comment by webdados** [comment]
Any developments on this?

**Comment by senadir** [comment]
No, I'm still a bit skeptic about adding a visual thing to the API, this can be 100% achieved with CSS thou.

**Comment by webdados** [comment]
I'm thinking in the line of the way to add custom fields to the classic checkout, in which you set that via the API (with css classes like form-row-first, form-row-last and form-row-wide, in fact 😏). [The fact that we can't add our own classes](https://github.com/woocommerce/woocommerce/issues/50795) to custom fields doesn't help to achieve this with our own CSS.

The idea is that WooCommerce could take care of the necessary styles on its own, agnostic from theme or custom fields implementation (like it does on the classic checkout). For example in Contact information and Additional information the CSS is not the same (not using flex) as in the Address fields, so the needed CSS for creating half sized fields is not the same.

We should make this easier IMHO, as this is not a simple edge case, but something people will start asking for when they start using this API more often.

**Comment by senadir** [comment]
> I'm thinking in the line of the way to add custom fields to the classic checkout, in which you set that via the API (with css classes like form-row-first, form-row-last and form-row-wide, in fact 😏).

Yes, that system was heavy on the UI side. The truth is, certain fields would be full width or half width depending on what's before them/after them, and screensize, and it will depend if whatever Checkout is rendering them respects those values or not.

**Comment by webdados** [comment]
I updated my reply above with additional information.

**Comment by pierorocca** [comment]
Reviewed from a product perspective and closing as not planned.

We are looking at how to handle this from an extensibility perspective. The current thinking is that the position of the additional field will dictate if it's half or full width. If it's dropped or placed into a row occupied by another field, they'll both occupy half width. For single fields on a row, full width. On mobile, always full width. For overrides, the right place for this is in the styling controls not in the API.
